app/src/main/java/com/asfoundation/wallet/topup/TopUpFragmentPresenter.kt

```diff
@@ -275,27 +292,9 @@ class TopUpFragmentPresenter(private val view: TopUpFragmentView,
     }
   }
 
-  private fun getChipIndex(inputValue: String): Int {
-    var index = -1
-    if (inputValue != "--") {
-      val value = inputValue.toBigDecimal()
-      for (i in chipValuesList.indices) {
-        if (value == chipValuesList[i].amount) {
-          index = i
-          break
-        }
-      }
-    }
-    return index
-  }
-
-  private fun getChipValuesList(): List<FiatValue> {
+  private fun getChipValue(index: Int): Single<FiatValue> {
     return interactor.getDefaultValues()
         .subscribeOn(networkScheduler)
```

Comment:
This subscribe logic should be on the chain that calls this and not here (it already is anyways)